### DISCUSSION TITLE
Maximum Density Gradients: Under-resolved shocks

### DISCUSSION THREAD

**Discussion by hwhitehead** [PrincetonUniversity/athena#507] [Q&A]
Perhaps a weird question, but does there exist a maximum density difference between cells beyond which Athena++ is unable to reliably solve?

I am attempting to resolve some very strong shocks and end up with a large density difference between cells. I believe this leads to issues with problematically large density/pressure gradients.

Has anyone else experienced this issue? Is there a combination of solver, reconstruction, integrator that people have found effective at dealing with potentially spatially under-resolved shocks?

**Comment by tomidakn** [comment]
This is actually a common problem in CFD. In general, any Eulerian FVM code can break down for flows with very high Mach numbers. This is a limitation of the scheme, so we need to face it, and should not push the code beyond the limit.

However, a certain configuration is more robust than others. Try
 - HLLE Riemann solver
 - PLM with minmod limiter (replace the van Leer limiters in plm.cpp by yourself)
 - Characteristic reconstruction (xorder = 2c)
The first two make the code more diffusive and robust. The characteristic reconstruction often provides a better reconstruction at waves with large amplitudes.

**Reply by hwhitehead** [reply]
Thanks for getting back to me, I will definitely try this.

**Reply by hwhitehead** [reply]
Do you know if there is any stability dependence on the time integration scheme?

**Reply by tomidakn** [reply]
I'm sorry but I don't know. RK2 and RK3 are strongly stability preserving schemes, but on the other hand VL2 uses the first order reconstruction in the predictor step, which may help the stability in some situations. I'd suggest you to test them. Also, try smaller time steps.
